modules/runner/src/runner.ts

```diff
@@ -130,6 +132,9 @@ export class Runner {
   private readonly pollIntervalMs: number;
   private readonly logIntervalMs: number;
   private pollInterval: ReturnType<typeof setInterval>;
+  private runningPromise: Promise<unknown>;
```

Comment:
```suggestion
  private runningPromise?: Promise<unknown>;
```

Since we `delete this.runningPromise` it is not only undefined sometimes, but it also an optional property of the class as it may not even be present on the object because of the `delete`. The type should reflect this for type safety.